### PR TITLE
Changes to allow for better error reporting

### PR DESCRIPTION
**Pull request by donasaur**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -68,13 +68,14 @@ MadMimi.prototype.request = function (method, route, data, done) {
...
+      username: self.username,
+      password: self.key,
+      err: JSON.stringify(err),
+      body: JSON.stringify(body)
     });
 
-    if (err || statusCode !== 200) {
-      done(errbot.initialize(statusCode, err || res.body));
-      return;
+    if (err || statusCode >= 400) {
```
Comment (indexzero) [inline lib/index.js]:
``` js
if (err || statusCode >= 400 || statusCode < 200) {
```

File: lib/index.js [hunk 2]
```diff
@@ -68,13 +68,14 @@ MadMimi.prototype.request = function (method, route, data, done) {
       uri: uri,
       statusCode: statusCode,
       fields: reqOpts.body || reqOpts.form,
-      err: err,
-      body: body
+      username: self.username,
+      password: self.key,
+      err: JSON.stringify(err),
```
Comment (jcrugzz) [inline lib/index.js]: stringifying an error doesn't always work unless its produced by `errs`. maybe just pass the message or explicitly enumerate properties here.

**Comment by indexzero** [comment]
+1 with one small fix. Since this is a major version bump (because of breaking change renaming `email` to `username`) you should consider adding a `CHANGELOG.md` file to document it.

**Comment by jcrugzz** [comment]
+1 with comment. Also agree with @indexzero on reporting changelogand bumping major

**Pull request merged**